Upml class container

UPML opened the pull request:
Первая часть первого дз. Гудкова Константина.

UPML committed:
add owner file

UPML committed:
add BeanXMLReader

UPML committed:
add firstPart of Graph

UPML committed:
some checkstyle fixes

UPML committed:
some more checkstyle

UPML commented:
Казалось бы уж в обработчике исключений имя e - это норма.

UPML commented:
Нужно проверять до коммита "add firstPart of Graph" дальше я начал реализовывать вторую часть задания.

UPML committed:
added Graph and Topsort

UPML committed:
fix reverse in topsort

UPML committed:
add 2*

arhangeldim commented:
СОгласен насчет имени исключения, поправил правила валидации, обновитесь с апстрима. Спасибо, о таком странном поведении сообщайте.

UPML committed:
want to rebuild in site.

UPML committed:
Merge remote-tracking branch 'upstream/master' into UPML-ClassContainer
fix checkout

UPML committed:
checkstyle

UPML commented:
Checkstyle побежден.

arhangeldim commented:
Задача принята, замечаний нет. Пока не закрывайте эту ветку, у задачи будет вторая часть.

Кстати, я вижу, что ваши билды фейлятся по чекстайлу на сервере. Но вы же можете локально это проверить без отправки на сервер - либо из консоли либо из идеи. На сервере запускается mvn clean install

UPML committed:
add task 3 and 4

UPML committed:
fix merge conflict
Merge remote-tracking branch 'upstream/master' into UPML-ClassContainer

Conflicts:
	src/main/java/arhangel/dim/container/Container.java

UPML committed:
fix merge conflict

UPML committed:
checkstyle

UPML committed:
more checkstyle

UPML committed:
more and more checkstyle

UPML committed:
Unit tests have appeared suddenly.

UPML committed:
added one empty line

UPML committed:
fix mistake

arhangeldim commented on src/main/java/arhangel/dim/container/BeanGraph.java:
```diff
@@ -10,41 +11,152 @@
+                dfs(tmp);
+            }
+        }
+        ArrayList<BeanVertex> reverseAnswer = new ArrayList<>();
```
Лучше слева (тип ссылки) указывать необходимый интерфейс, в данном случае List

arhangeldim commented on src/main/java/arhangel/dim/container/Container.java:
```diff
@@ -1,29 +1,102 @@
+        try {
+            object = clazz.newInstance();
+        } catch (InstantiationException e) {
+            System.err.println("Can't initialize class " + e.getMessage());
```
Мне кажется лучше пробросить new InvalidConfigurationException(e);
А на уровень выше, там где этот метод вызывается, его обработать

arhangeldim commented on src/main/java/arhangel/dim/container/Container.java:
```diff
@@ -1,29 +1,102 @@
+                    method.invoke(object, new Integer(property.getValue()));
+                }
+            } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+                System.err.println(e.getMessage());
```
Также, лучше пробросить исключение наверх

roller145 commented on src/main/java/arhangel/dim/container/Container.java:
```diff
@@ -1,29 +1,102 @@
+            try {
+                if (property.getType() == ValueType.REF) {
+                    Object parameter = getByName(property.getValue());
+                    Method method = clazz.getMethod(methodName,
```
возможно, стоит создать поле:
Field field = null;
            try {
                field = clazz.getDeclaredField(pr.getName());
                
            } catch (NoSuchFieldException e) {
                System.err.println("incorrect name of class " + e.getMessage());
                return;
            }
